File: trunk/acm/cc/spoj27.cpp
/**
 * @file    27.cpp
 * @author  FAN Kai (fankai@net.pku.edu.cn), Peking University 
 * @date    May 20 21:40:30 CST 2009
 */

#include <iostream>
#include <vector>
#include <string>
using namespace std;

char line[100];

int main() {
    int n, m; 
    cin >>n;
    for (int i = 0; i < n; ++i) {
        vector<string> vs;
        cin >>m;
        cin.getline(line, 99);
        for (int j = 0; j < m; ++j) {
            cin.getline(line, 99);
            vs.push_back(line);
        }
        sort(vs.begin(), vs.end());
        string last = vs[0];
        int count = 1;
        for (int j = 1; j < m; ++j) {
            if (vs[j] == last)
                ++count;
            else {
                cout <<last <<" " <<count <<endl;
                last = vs[j];
                count = 1;
            }
        }
        cout <<last <<" " <<count <<endl;
        cout <<endl;
    }
    return 0;
}

File: acm/python/spoj27.py
#!/usr/bin/env python
# @author   FAN Kai (fankai@net.pku.edu.cn), Peking University  
# @date     May 20 21:54:45 CST 2009

def mainflow():
    n = input()
    for i in range(n):
        if i > 0: raw_input()
        m = input()
        alist = []
        for j in range(m):
            line = raw_input()
            alist.append(line)
        alist.sort()

        last = alist[0]
        count = 0

        for j in range(1, m):
            if alist[j] == last:
                count += 1
            else:
                print last, count
                last = alist[j]
                count = 1
        print last, count
        print

mainflow()



